EducationalSystem.WebAPI/Controllers/CoursesController.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Mail;
+using System.Threading.Tasks;
+using AutoMapper;
+using DatabaseStructure;
+using DatabaseStructure.Models;
+using EducationalSystem.WebAPI.ViewModels;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace EducationalSystem.WebAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CoursesController : ControllerBase
+    {
+        DBContext db;
+        DataManager dataManager;
+
+        public CoursesController(DBContext context)
+        {
+            db = context;
+            dataManager = new DataManager(db);            
+        }
+
+        [HttpPost("{action}")]
+        public ActionResult<Course> AddCourse([FromBody]Course course)
+        {            
+            if (course == null)
+            {
+                return NotFound();
+            }            
+            db.Courses.Add(course);            
+            var professor = db.Professors.FirstOrDefault(p => p.Id == course.ProfessorId);           
+            course.Professor = professor;            
+            db.SaveChanges();
+            var school = db.Schools.FirstOrDefault(s => s.Id == professor.SchoolId);
+            var config = new MapperConfiguration(cfg => cfg.CreateMap<Course, CreateCourseViewModel>()
```

Comment:
Please create one mapper profile for each entity to avoid create them always in methods, for example
![image](https://user-images.githubusercontent.com/4987541/98829810-b18e3980-244a-11eb-8e7a-40ba1385c76a.png)